src/test/unit/test_spans_container.py

```diff
@@ -83,6 +78,23 @@ def test_spans_container_end_function_send_only_on_errors_mode_false_not_effecti
     assert reported_ttl is not None
 
 
+def test_spans_container_timeout_mechanism_send_only_on_errors_mode(
+    monkeypatch, context, reporter_mock, dummy_http_request
+):
+    monkeypatch.setattr(Configuration, "send_only_if_error", True)
+
+    SpansContainer.create_span()
+    SpansContainer.get_span().start()
+    SpansContainer.get_span().add_request_event(dummy_http_request)
+
+    SpansContainer.get_span().handle_timeout()
+
+    messages = reporter_mock.call_args_list[0][1]["msgs"]
```

Comment:
If you can find a better way to extract the messages, it will be nice :)
Maybe you can use the pattern we have in testing-utils. Not a must.